packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -525,6 +502,48 @@ contract RoutersFacet is BaseConnextFacet {
 
   // ============ Internal functions ============
 
+  /**
+   * @notice Sets the router recipient
+   * @param _router The router to set the recipient for
+   * @param _updated The recipient to set
+   * @param _previous The existing recipient
+   */
+  function _setRouterRecipient(
+    address _router,
+    address _updated,
+    address _previous
+  ) internal {
+    // Check recipient is changing
+    if (_previous == _updated) revert RoutersFacet__setRouterRecipient_notNewRecipient();
+
+    // Set new recipient
+    s.routerConfigs[_router].recipient = _updated;
+
+    // Emit event
+    emit RouterRecipientSet(_router, _previous, _updated);
+  }
+
+  /**
+   * @notice Sets the router owner
+   * @param _router The router to set the owner for
+   * @param _updated The owner to set
+   * @param _previous The existing owner
+   */
+  function _setRouterOwner(
+    address _router,
+    address _updated,
+    address _previous
+  ) internal {
+    // Check owner is changing
+    if (_previous == _updated) revert RoutersFacet__setRouterOwner_noChange();
+
+    // Set new owner
+    s.routerConfigs[_router].owner = _updated;
+
+    // Emit event
+    emit RouterOwnerAccepted(_router, _previous, _updated);
+  }
+
```

Comment:
Need to sanity check that the router has approval in `_addLiquidityForRouter`! If the router is unapproved and liquidity is added, then that will stack against our liq caps w/o actually offering up liquidity.

Reply:
that check exists (L550-552):

```js
// Sanity check: router is approved.
if (!_isRouterWhitelistRemoved() && !getRouterApproval(_router))
    revert RoutersFacet__addLiquidityForRouter_badRouter();
```